---
title: "GDSCN Book: SARS with Galaxy on AnVIL"
date: "`r format(Sys.time(), '%B %d, %Y')`"
site: bookdown::bookdown_site
documentclass: book
bibliography: book.bib
biblio-style: apalike
link-citations: yes
description: "This book provides resources for instructors to engage students in a cloud-based Galaxy activity on AnVIL, focused on SARS-CoV-2 variant detection."
favicon: assets/anvil_favicon.ico
output:
    bookdown::word_document2:
      reference_docx: assets/gdscn-template.docx
      toc: true
---

# Overview {-}

This book provides resources for instructors to engage students in a cloud-based Galaxy activity on AnVIL, focused on SARS-CoV-2 variant detection.

There is a growing need for undergraduate students to learn cutting-edge concepts in genomics data science, including performing analysis on the cloud instead of a personal computer. This lesson aims to introduce a mutant detection bioinformatics pipeline based on a publicly available genetic sample of SARS-CoV-2. Students will be introduced to the sequencing revolution, variants, genetic alignments, and essentials of cloud computing prior to the lab activity. During the lesson, students will work hands-on with the point-and-click Galaxy interface on the [AnVIL](https://anvilproject.org/) cloud computing resource to check data, perform an alignment, and visualize their results.

## Skills Level {-}

:::{.notice}
_Genetics_  
**Novice**: no genetics knowledge needed
 
_Programming skills_  
**Novice**: no programming experience needed
:::

## Learning Objectives {-}

Learning objectives for this activity come from the [Genetics Core Competencies](https://genetics-gsa.org/education/genetics-learning-framework/):

- Gather and evaluate experimental evidence, including qualitative and quantitative data
- Generate and interpret graphs displaying experimental results
- Critique large data sets and use bioinformatics to assess genetics data
- Tap into the interdisciplinary nature of science

## AnVIL Collection {-}

Please check out our full collection of AnVIL resources below!

```{r, echo = FALSE, message = FALSE, warning = FALSE}
source("scripts/anvil_repo_table.R")

# Add below any repositories that should not be added
exclude = c(
  "anvil-intro",
  "AnVIL_Template",
  "AnVIL_Template_Test",
  "GDSCN_bookdown_style"
)

# Create table for AnVIL repos detected through the API
knitr::kable(make_anvil_repo_table(exclude), format = "html")
```
